RELEASE.md

```diff
@@ -1,88 +1,37 @@
-
-The SRI repo holds two principal branches with an infinite lifetime:
-- `main`
-- `dev`
-
-We consider `main` to be the branch where the source code of `HEAD` always reflects the latest release.
-
-We consider `dev` to be the branch where the source code of `HEAD` always reflects the latest changes in the development cycle.
-
-The SRI team will decide the appropriate time when the changes to `dev` are merged back into `main` and then tagged with a release number while bumping `MAJOR`, `MINOR`, or `PATCH`.
-
-## Feature Branches
-
-New features are developed into separate branches that only live in the contributor's forks.
-
-- branch off from: `dev`
-- merge back into: `dev`
-- naming convention: `feat-x`, where `x` describes the feature
-
-## Patch Branches
-
-Bugs are patched into separate branches that only live in the contributor's forks.
-
-- branch off from: `dev`
-- merge back into: `dev`
-- naming convention: `patch-x`, where `x` describes the bug/patch
-
-## Release Branches
-
-Every new release, a branch is created.
-
-- branch off from: `main`
-- naming convention: `vMAJOR.MINOR.PATCH`
-
-# Releasing Roles Binaries
-
-The [release page of SRI repo](https://github.com/stratum-mining/stratum/releases) provides executable binaries for all SRI roles, targeting popular hardware architectures.
-
-The GitHub binary releases of the roles are handled in the `release-bin.yaml` workflow.
-
-This workflow is manually started by navigating to the "Actions" tab in the SRI repo, then navigating to the Release workflow section, and clicking "Run Workflow".
-
-Every time the workflow is manually triggered, the correct release branch must be chosen.
-
-Note: in order to be able to manually trigger the "Run Workflow" button, the user needs to have "Write" permissions on the repository, otherwise the button will not show up on the UI.
-
-# Publishing Library Crates
-
-Although SRI has a global release cycle, which is attached to the binaries, each internal crate also has its own versioning history.
-
-Lib crates are published to crates.io in the `release-lib.yaml` workflow. The workflow tries to update all the library crates. 
-
-If a crate is not updated successfully, the step will fail, but since all steps have `continue-on-error` set to true, the workflow will continue.
-
-Since steps can fail, the output of the action must be manually checked.
-
-Every PR to `main` needs to increase the version of whatever crate it is touching. Otherwise, we will mess up the dependency chain of whoever is fetching from crates.io
-
-Manually running `cargo release` in the various workspaces helps to prepare the version number before the releases are published.
-
-Every time we bump some crate's version, `release-libs.yaml` needs to be manually triggered in order to update crates.io.
+## Release Process
+
+- Constrain each development cycle to a fixed time period, after which a release
+  is made. Typically, the development cycle is 6-8 weeks long.
```

Comment:
@pavlenex @GitGab19  what are your thoughts about having a time constrain on release cycles?

Reply:
I think that if we release ONLY on 6/8 weeks we will find ourself rushing to get PR merged nearby the release. That of course will impact the quality of the code. For example if we have a feature that we need and is almost ready and release is in 3 days we will do everything possible to have it in the next release otherwise we will have to wait 8 weeks.

Reply:
I think that we should aim to a more precise time frame for future releases (even if it's generally hard in a FOSS environment), and 6 to 8 weeks looks reasonable to me. If we will encounter some barriers as the one mentioned by @Fi3 I'm pretty sure we could postpone the specific release a little bit. At the end we will keep asserting the progress on every dev call as usual. As a general guideline, I would leave it

Reply:
IMHO having shorter release cycle solve better this issue

Reply:
I'm not too opinionated on this, but is seems a middle ground would be to:
- aim for 4/6 weeks (shorter cycle)
- be open to exceptions in case specific PRs are taking longer than expected, in which case we postpone those specific PRs to the next release

what do you guys think?

Reply:
I would wait for @pavlenex input, he's definitely the one with more experience regarding time frames and roadmap definition in FOSS 😉

Reply:
> if some specific task is taking longer than expected, we postpone it to the next release (so that we're not rushing anything)

yep this is ideal and should be always like that but there is a positive correlation between the length of release cycle and the willinges to ignore best practice and rushing thing to not miss the release.

Reply:
it seems @pavlenex won't be available to provide his input

for now, let's avoid blocking this PR because of this

let's just remove this phrase about release timeline

Reply:
> it seems @pavlenex won't be available to provide his input
> 
> for now, let's avoid blocking this PR because of this
> 
> let's just remove this phrase about release timeline

Agree 👍